tests/src/Unit/SocialAuthUserTest.php

```diff
@@ -0,0 +1,1138 @@
+<?php
+
+namespace Drupal\Tests\social_auth\Unit;
+
+use Drupal\Core\Config\ConfigFactoryInterface;
+use Drupal\Core\Entity\EntityFieldManagerInterface;
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\Core\File\FileSystemInterface;
+use Drupal\Core\Language\LanguageManagerInterface;
+use Drupal\Core\Logger\LoggerChannelFactoryInterface;
+use Drupal\Core\Logger\LoggerChannelInterface;
+use Drupal\Core\Messenger\MessengerInterface;
+use Drupal\Core\Routing\RouteProviderInterface;
+use Drupal\Core\Session\AccountProxyInterface;
+use Drupal\Core\Transliteration\PhpTransliteration;
+use Drupal\Core\Utility\Token;
+use Drupal\social_auth\SocialAuthDataHandler;
+use Drupal\social_auth\User\SocialAuthUser;
+use Drupal\social_auth\User\UserAuthenticator;
+use Drupal\social_auth\User\UserManager;
+use Drupal\Tests\UnitTestCase;
+use Drupal\user\Entity\User;
+use Drupal\user\UserInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Symfony\Component\EventDispatcher\EventDispatcherInterface;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+
+/**
+ * Tests social_auth User.
+ *
+ * @group social_auth
+ */
+class SocialAuthUserTest extends UnitTestCase {
+
+  /**
+   * The tested Social Auth User.
+   *
+   * @var \Drupal\social_auth\User\SocialAuthUser|\PHPUnit_Framework_MockObject_MockObject
+   */
+  protected $socialAuthUser;
+
+  /**
+   * The tested Social Auth UserManager.
+   *
+   * @var \Drupal\social_auth\User\UserManager|\PHPUnit_Framework_MockObject_MockObject
+   */
+  protected $userManager;
+
+  /**
+   * The tested Social Auth UserAuthenticator.
+   *
+   * @var \Drupal\social_auth\User\UserAuthenticator
+   */
```

Comment:
Don't forget about `|\PHPUnit_Framework_MockObject_MockObject`

Reply:
You have to repeat the process with all mocked objects. The point is that, for instance, my text editor highlights things as an error because you call the method expects which is only defined in the `\PHPUnit_Framework_MockObject_MockObject` class